lindi/conversion/attr_conversion.py

```diff
@@ -0,0 +1,162 @@
+from typing import Any, Union
+import numpy as np
+import h5py
+from .nan_inf_ninf import encode_nan_inf_ninf
+from .h5_ref_to_zarr_attr import h5_ref_to_zarr_attr
+
+
+def h5_to_zarr_attr(attr: Any, *, label: str = '', h5f: Union[h5py.File, None]):
+    """Convert an attribute from h5py to a format that zarr can accept."""
+
+    from ..LindiH5pyFile.LindiH5pyReference import LindiH5pyReference  # Avoid circular import
+
+    # Do not allow these special strings in attributes
+    special_strings = ['NaN', 'Infinity', '-Infinity']
+
+    if isinstance(attr, list):
+        list_dtype = _determine_list_dtype(attr)
+        attr = np.array(attr, dtype=list_dtype)
+
+    if isinstance(attr, str) and attr in special_strings:
+        raise ValueError(f"Special string {attr} not allowed in attribute value at {label}")
+    if attr is None:
+        raise Exception(f"Unexpected h5 attribute: None at {label}")
+    elif type(attr) in [int, np.int8, np.int16, np.int32, np.int64, np.uint8, np.uint16, np.uint32, np.uint64]:
+        return int(attr)
+    elif type(attr) in [float, np.float16, np.float32, np.float64, np.float128]:
+        return encode_nan_inf_ninf(float(attr))
+    elif type(attr) in [complex, np.complex64, np.complex128, np.complex256]:
```

Comment:
```suggestion
    elif isinstance(attr, complex) or (isinstance(attr, np.ndarray) and np.issubdtype(attr.dtype, np.complex)):
```

`np.complex256` also doesn't exist on my system